src/installer/image/installer.go

```diff
@@ -37,17 +45,57 @@ func GetDigest(uri string) (string, error) {
 }
 
 func NewImageInstaller(fs afero.Fs, props *Properties) installer.Installer {
+	// Create default transport
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+
+	if props.Dynakube.HasProxy() {
+		proxy, err := props.Dynakube.Proxy(context.TODO(), props.ApiReader)
+		if err != nil {
+			log.Info("failed to get proxy from dynakube", "proxy", proxy)
+			return nil
+		}
+
+		proxyUrl, err := url.Parse(proxy)
+		if err != nil {
+			log.Info("invalid proxy url", "proxy", proxy)
+		}
+		log.Info("proxy spec", "proxy", proxy, "proxyURL.Host", proxyUrl.Host, "proxyURL.Port()", proxyUrl.Port())
+
+		transport.Proxy = func(req *http.Request) (*url.URL, error) {
+			return proxyUrl, nil
+		}
+	}
+
+	if props.Dynakube.Spec.TrustedCAs != "" {
+		trustedCAs, err := props.Dynakube.TrustedCAs(context.TODO(), props.ApiReader)
+		if err != nil {
+			return nil
+		}
+
+		rootCAs := x509.NewCertPool()
+		if ok := rootCAs.AppendCertsFromPEM(trustedCAs); !ok {
+			log.Info("failed to append custom certs!")
+		}
+
+		if transport.TLSClientConfig == nil {
+			transport.TLSClientConfig = &tls.Config{} //nolint:gosec
+		}
+		transport.TLSClientConfig.RootCAs = rootCAs
+	}
+
```

Comment:
We could think of reusing the function `addCertificates` from `image_hash.go`? Up to you, but the code is the same.

Reply:
yea, but it's private function also inside controllers, which I don't want to introduce this dependency to installer which is independent from operator.

[A little copying is better than a little dependency.](https://www.youtube.com/watch?v=PAAkCSZUG1c&t=9m28s)

Reply:
:laughing: :laughing: :laughing: